### PR TITLE
Add integration tests and add cancel + wait functionality

### PR DESCRIPTION
**Pull request by ericliuche**
- Adds integration tests
- First send SIGTERM then SIGKILL in `Cancel()`
- Refactor execution id -> cmd executor logic into `CmdExecutorManager`
- Change behavior of `/cancel` so that a cancel without an execution id is only valid if there is one execution

### CODE REVIEW

**Review by zhan-airplane — COMMENTED** [review]
I like the tests :) A couple of questions about some cases first though

File: pkg/httpd/server_test.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+		)
+	}
+	wg.Add(3)
+	go f()
+	go f()
+	go f()
+	wg.Wait()
+}
+
+func TestExecuteCmdArgsAndEnv(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (ericliuche) [reply pkg/httpd/server_test.go:207]: ah yup

File: pkg/httpd/server_test.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	body := h.POST("/").
+		WithJSON(ExecutCmdReq{
+			Args: []string{"--non-existing-flag"},
+		}).
+		Expect().
+		Status(http.StatusOK).Body()
+
+	outputs := parseOutputs(body.Raw(), require)
+	require.Equal(
+		Output{Msg: "exit status 2", OutputType: OutputTypeExitStatus, IsError: true},
```
[the remark this answers is not in the document]
Comment (ericliuche) [reply pkg/httpd/server_test.go:135]: yeah, i added the `IsError` field for that. should it be something different?

[the remark this answers is not in the document]
Comment (zhan-airplane) [reply pkg/httpd/server_test.go:135]: Oh oops haha I misinterpreted that, that's fine.

Actually what do you think of making it a field like `Status` instead? and have two states `Success` and `Failed` for now, but with the option to add more later if we need to handle stranger cases

**Review by zhan-airplane — APPROVED** [review]

File: pkg/httpd/cmdexecutor.go [hunk 3]
```diff
@@ -1,35 +1,142 @@
...
+	// Delete even if the process doesn't exist.
+	if cmdExecutor.hasActiveProcess() {
+		return errors.New("unable to delete active cmd, must cancel first")
+	}
+	defer delete(c.Executors, executionID)
+	return nil
+}
+
+func (c *CmdExecutorManager) CancelExecutor(executionID string) error {
+	c.Mutex.Lock()
```
[the remark this answers is not in the document]
Comment (ericliuche) [reply pkg/httpd/cmdexecutor.go:74]: yeah i like that

**Pull request merged**